plugin/trino-redshift/src/test/java/io/trino/plugin/redshift/TestRedshiftUnload.java

```diff
@@ -0,0 +1,195 @@
+import io.trino.testing.sql.TestTable;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.parallel.Execution;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.NoSuchKeyException;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static com.google.common.collect.MoreCollectors.onlyElement;
+import static io.trino.plugin.redshift.RedshiftQueryRunner.IAM_ROLE;
+import static io.trino.plugin.redshift.TestingRedshiftServer.TEST_SCHEMA;
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static io.trino.testing.TestingProperties.requiredNonEmptySystemProperty;
+import static io.trino.tpch.TpchTable.NATION;
+import static java.util.Locale.ENGLISH;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+import static org.junit.jupiter.api.parallel.ExecutionMode.CONCURRENT;
+
+@TestInstance(PER_CLASS)
+@Execution(CONCURRENT)
+final class TestRedshiftUnload
+        extends AbstractTestQueries
+{
+    private static final String S3_UNLOAD_ROOT = requiredNonEmptySystemProperty("test.redshift.s3.unload.root");
+    private static final String AWS_REGION = requiredNonEmptySystemProperty("test.redshift.aws.region");
+    private static final String AWS_ACCESS_KEY = requiredNonEmptySystemProperty("test.redshift.aws.access-key");
+    private static final String AWS_SECRET_KEY = requiredNonEmptySystemProperty("test.redshift.aws.secret-key");
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return RedshiftQueryRunner.builder()
+                .setConnectorProperties(
+                        Map.of(
+                                "redshift.unload-location", S3_UNLOAD_ROOT,
+                                "redshift.unload-iam-role", IAM_ROLE,
+                                "s3.region", AWS_REGION,
+                                "s3.aws-access-key", AWS_ACCESS_KEY,
+                                "s3.aws-secret-key", AWS_SECRET_KEY))
+                .setInitialTables(List.of(NATION))
+                .build();
+    }
+
+    @Test
+    void testUnloadEnabled()
+    {
+        assertQuery(
+                "SHOW SESSION LIKE 'redshift.unload_enabled'",
+                "VALUES ('redshift.unload_enabled', 'true', 'true', 'boolean', 'Use UNLOAD for reading query results')");
+    }
+
+    @Test
+    void testUnload()
+    {
+        assertQueryStats(
+                getSession(),
+                "SELECT nationkey, name FROM nation WHERE regionkey = 0",
```

Comment:
Test would be better if it was doing a UNION of two scans on redshift so that we exercise the logic for different output path per scan.

Reply:
ya good idea with the UNION.